src/plugins/intel_gpu/src/graph/reorder.cpp

```diff
@@ -286,8 +286,8 @@ void reorder_inst::update_output_memory() {
 
     // Can_be_optimized nodes are allocating from memory_pool too. In this case,
     // we need release the legacy output memory from memory pool explicitly.
-    if (static_cast<bool>(_outputs[0]) &&
-        _node->get_program().get_config().get_enable_memory_pool()) {
+    // Ensure the output layout is static in case the output memory is reinterpreted with a dynamic layout
+    if (static_cast<bool>(_outputs[0]) && _outputs[0]->get_layout().is_static() && _node->get_program().get_config().get_enable_memory_pool()) {
```

Comment:
but in this case, this memory is still not released right? while actually it can
it will be good if we can fix this issue, meanwhile, we can keep the memory optimization as well.
otherwise, there are some other nodes (like reshape, strided-slice), not only reorder, also have potential issue